src/Data/HugeNum.purs

```diff
@@ -73,6 +74,14 @@ timesSign Plus Plus = Plus
 timesSign Minus Minus = Plus
 timesSign _ _ = Minus
 
+instance isForeignHugeNum :: IsForeign HugeNum where
+  read f = do
+    n <- readNumber f
+    pure (fromNumber n)
+
+instance asForeignHugeNum :: AsForeign HugeNum where
+  write v = write (toNumber v)
```

Comment:
Yeah, I think generally we need to improve the documentation of `Foreign`, as it's not clear that it's intended for interop with JS, not arbitrary serialization of JS and JSON / JSON parsing.

I think making foreign have laws in the usual sense would be a bit tricky, but yeah, definite room for improvement.

Going way off topic here, but I guess the best way I can describe it is this: when using `Foreign` there should be no change at all in the representation of the value (null/undefined being maybe-ified aside), it's there to verify a value fits an expected representation.